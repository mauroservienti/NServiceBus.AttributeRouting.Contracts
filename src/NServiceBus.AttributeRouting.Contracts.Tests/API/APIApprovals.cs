using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;

namespace NServiceBus.AttributeRouting.Contracts.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(RouteAttribute).Assembly.GeneratePublicApi(new ApiGeneratorOptions(){
                ExcludeAttributes =
                [
                    "System.Runtime.Versioning.TargetFrameworkAttribute"
                ]
            });

            Approvals.Verify(publicApi.Replace(".git", ""));
        }
    }
}