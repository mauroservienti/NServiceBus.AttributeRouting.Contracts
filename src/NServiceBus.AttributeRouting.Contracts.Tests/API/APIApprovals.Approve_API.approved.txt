[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "https://github.com/mauroservienti/NServiceBus.AttributeRouting.Contracts")]
namespace NServiceBus.AttributeRouting.Contracts
{
    [System.AttributeUsage(System.AttributeTargets.Assembly)]
    public class RouteAttribute : System.Attribute
    {
        public RouteAttribute(string commandsTo = null, string messagesTo = null) { }
        public string CommandsDestination { get; }
        public string MessagesDestination { get; }
    }
    [System.AttributeUsage(System.AttributeTargets.Class, Inherited=false)]
    public class RouteToAttribute : System.Attribute
    {
        public RouteToAttribute(string destination) { }
        public string Destination { get; }
    }
}